### ISSUE TITLE
& is detected as special character but not escaped in some cases

### ISSUE THREAD

**Issue by level44** [cackharot/suds-py3#33]
& is special/reserved character for the XML so should be escaped in the all cases.
Currently suds-py3 library detects that data need to be escaped but does not escape &. It is reproducible with a following content:
- `&amp;`
- `&lt;`
- `&gt;`
- `&quot;`
- `&apos;`

For example according to the XML specification string "`&lt;`" shall be escaped to "`&amp;lt;`".
Please refer to online escapers like https://www.freeformatter.com/xml-escape.html  or example written in golang:

```
//go 1.6.2

package main  
import (
    "fmt"
    "encoding/xml"
    "bytes"
)
	
func main() { 
    var buf bytes.Buffer
    xml.Escape(&buf,[]byte("&lt;test"))
    fmt.Printf(buf.String()) // output here is &amp;lt;test
}
```

Please also look at this example (needsEncoding returns True):
```
import suds
encoder = suds.sax.Encoder()
encoder.needsEncoding('&lt;test') # output is True
encoder.encode('&lt;test') # output is '&lt;test' but should be &amp;lt;
```

It is limited in the sax/enc.py file:

```    
encodings = (
        ('&(?!(amp|lt|gt|quot|apos);)', '&amp;'),
        ('<', '&lt;'),
        ('>', '&gt;'),
        ('"', '&quot;'),
        ("'", '&apos;')
    )
```

**Comment by cackharot** [comment]
@level44 
It is weird but the original authors have specifically not encoding again the already encoded string, by doing a negative lookahead `'&(?!(amp|lt|gt|quot|apos);)'` since I don't have enough context on why they did it that way, it's hard to refactor/fix now.

But we can take a leap and remove the negative lookahead regex, but not sure in which situation it will break.

Let me know what you think

**Comment by cackharot** [comment]
Released 1.4.0.0 with this fix + others.
